src/clients/vendor.rb

```diff
@@ -98,18 +98,15 @@ def main
       else
         Pkg.TargetInit("/", false)
 
-        @products = Pkg.ResolvableProperties("", :product, "")
+        @products = Y2Packager::Resolvable.find(kind: :product,
+           status:    :installed,
+           category:  "base")
         @products = [] if @products == nil
-        @products = Builtins.filter(@products) do |p|
-          Ops.get(p, "status") == :installed
-        end
-        @base = Builtins.filter(@products) do |p|
-          Ops.get_string(p, "category", "") == "base"
-        end
         @base = deep_copy(@products) if Builtins.size(@base) == 0
-        @product = Ops.get(@base, 0, {})
-        @version = Ops.get_string(@product, "version", "")
-        @version = Ops.get(Builtins.splitstring(@version, "-"), 0, "") # split off release
+        @product = @base[0]
+        if @product
```

Comment:
The result of this code might be slightly different. In the original case, if `@products` is empty, `@version` will be `""`. However, in the new code, `@version` will be `nil`. Or am I wrong?